### PR TITLE
[feature] Account info now says if it's missing a required field.

### PR DESCRIPTION
**Pull request by Johnetordoff**
**Purpose**
Give more useful error message if required field is left blank. Close Issue #3434.
**Changes**
Now check to see if error is caused by missing field and changes error message accordingly.
**Side Effects**
None that I know of.

### CODE REVIEW

File: website/static/js/profile.js [hunk 1]
```diff
@@ -264,6 +264,13 @@ BaseViewModel.prototype.handleSuccess = function() {
 BaseViewModel.prototype.handleError = function(response) {
     var defaultMsg = 'Could not update settings';
     var msg = response.message_long || defaultMsg;
+
+    for(var i = 0 ;i<this.contents().length; i++){
+        if(this.contents()[i].institution() === ''){
```
Comment (sloria) [inline website/static/js/profile.js]: This doesn't feel right; the `BaseViewModel` shouldn't have to know about its subclasses' fields (e.g. `institution`). Can you use knockout-validation on the institution field instead? Or expose a hook to subclasses that allows them to perform additional validation?

**Comment by Johnetordoff** [comment]
@lyndsysimon I've been really puzzling over how to fix this, but it seems like it should be a quick fix given the validation already in place. Any thoughts?

**Comment by Johnetordoff** [comment]
@lyndsysimon This is done, it was a quick fix after all.

**Comment by Johnetordoff** [comment]
@lyndsysimon Now this is <em>really</em> done.

File: website/templates/include/profile/schools.mako [hunk 2]
```diff
@@ -29,9 +29,9 @@
 
                     <div class="form-group">
                         <label>Institution</label>
-                        <input class="form-control" data-bind="value: institution"
```
Comment (lyndsysimon) [inline website/templates/include/profile/schools.mako]: This line was un-indented a single space - it no longer matches the level above it.

**Comment by lyndsysimon** [comment]
No changes required - @sloria, this one is all yours :)

**Comment by lyndsysimon** [comment]
As a general comment - there are a few superfluous lines that have been added in your PR, and a couple of places where whitespace has been modified. I believe this was simply overlooked.

To avoid this, I recommend adding code to a pending commit using the command `git add -p .`. This will give you the opportunity to review every line and character that gets added to the commit - I use this command almost exclusively, to the point that I've aliased it in my shell to simply `g a`.  Entering `?` in the interactive prompt it gives you will print a list of commands and their definitions.

**Comment by sloria** [comment]
Good advice, @lyndsysimon .

This has already been merged into release/0.43.0

**Pull request closed without merging**